flexibuzz-cpp/flexibuzz.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef FLEXIBUZZ_H
+#define FLEXIBUZZ_H 1
+
+#include <functional>
+#include <map>
+
+template<class K, class V>
+class flexibuzz
+{
+public:
+
+    flexibuzz(
+        std::function<bool(const K&, const K&)> match,
+        std::function<V(const K&)> map,
+        std::map<K,V> values) :
+            _match(match),
+            _map(map),
+            _values(values)
+    {}
+
+    V operator[] (const K& k) const
+    {
+        for (auto it = _values.crbegin(); it != _values.rend(); ++it)
```

Comment:
Add terrible type safety hack I sent you ;).

Reply:
I probably just don't understand, but it seems like it doesn't work

https://godbolt.org/z/lF7iaw